### PR TITLE
Test

### PR DESCRIPTION
**Pull request by Ab5oluteG**
added command and operation test files in test directory + minor changes to command.py and operation.py mainly in init_to_dict functions and lead functions respectively (these functions were not being used before so they werent fully fleshed out),

Added python-test.yml for github actions to trigger on main and check for tests errors, python black formatting errors, and pylint errors.

### CODE REVIEW

File: src/agent/config/operation.py [hunk 1]
```diff
@@ -109,15 +109,16 @@ def get_settings(self) -> dict:
         Gets the settings of the operation
         :return: a dict representing the operations
         """
-        super().get_settings()["headless"] = self.headless
-        return super().get_settings()
+        settings = super().get_settings()
+        settings["headless"] = self.headless
+        return settings
```
Comment (govindansriram) [inline src/agent/config/operation.py:114]: why did you change this?

File: src/agent/config/operation.py [hunk 2]
```diff
@@ -154,13 +155,18 @@ class LLMSettings(dict):
     A dictionary subclass for the settings of an LLM.
     """
 
-    def __init__(self, name: Union[str, None] = None, api_key: Union[str, None] = None):
+    def __init__(
+        self, name: Union[str, None] = None, api_key: Union[str, None] = None, **kwargs
+    ):
+        # pylint: disable=W0613
```
Comment (govindansriram) [inline src/agent/config/operation.py:161]: what are we disabling here

File: src/agent/config/operation.py [hunk 3]
```diff
@@ -265,12 +271,21 @@ def get_settings(self):
...
+        llm_opts = cls(
+            try_limit=data_dict["settings"]["try_limit"],
+            timeout=data_dict["settings"]["timeout"],
+            max_tokens=data_dict["settings"]["max_tokens"],
+            llm_settings=[
+                LLMSettings(**llm_setting)
+                for llm_setting in data_dict["settings"]["llm_settings"]
+            ],
+            workflow_type=data_dict["settings"]["workflow"]["workflow_type"],
+        )
```
Comment (govindansriram) [inline src/agent/config/operation.py:288]: Good job I like the way you did this

File: test/agent/config/command_test.py [hunk 4]
```diff
@@ -0,0 +1,775 @@
...
+
+
+def test_command_init():
+    """
+    Function to test initialization of Command
+    """
+    command = Command("llm", "test_command", {"param": "value"})
+    assert command.command_type == "llm"
+    assert command.command_name == "test_command"
+    assert command.params == {"param": "value"}
```
Comment (govindansriram) [inline test/agent/config/command_test.py:37]: I dont know if you need to test it this much. Mainly just function values, its good either way but not nessecary

File: .github/workflows/python-test.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      run: poetry install
+
+    - name: Run Tests
+      run: poetry run pytest test/agent
+
+    - name: Check Python Black formatting
+      run: poetry run black --check .
+
+    - name: Run Pylint
+      run: poetry run pylint src/agent/config test/agent/config
```
[the remark this answers is not in the document]
Comment (Ab5oluteG) [reply .github/workflows/python-test.yml:38]: thats my b, copied it from the next section just supposed to be src agent

File: src/agent/config/operation.py (hunk 1 above)
Comment (Ab5oluteG) [reply src/agent/config/operation.py:114]: it wasn't working with how it was originally. doing this:
super().get_settings()["headless"] = self.headless
        return super().get_settings()
was returning an empty string

File: src/agent/config/operation.py (hunk 2 above)
Comment (Ab5oluteG) [reply src/agent/config/operation.py:161]: **kwargs is an unused argument but because of the way i set up llm_opts in in line 279-288, more kwargs were being passed in so i needed to allow that in the this init so it wouldn't give a linting error. its on the next line to keeo it in the scope of that function

File: .github/workflows/python-test.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (govindansriram) [reply .github/workflows/python-test.yml:38]: ok Ill resolve this when you push the changes

File: src/agent/config/operation.py (hunk 2 above)
Comment (govindansriram) [reply src/agent/config/operation.py:161]: hmmm are you sure you need kwargs?

**Review by govindansriram — COMMENTED** [review]
looks good so far resolve the other issues and its a go ahead

File: src/agent/config/operation.py (hunk 2 above)
Comment (Ab5oluteG) [reply src/agent/config/operation.py:161]: i think so cuz i get this error when i remove it
TypeError: LLMSettings.__init__() got an unexpected keyword argument 'model'

Comment (Ab5oluteG) [reply src/agent/config/operation.py:161]: it shouldn't impact functionality since its not used in the function and its optional anyways

**Review by govindansriram — APPROVED** [review]
